src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs

```diff
@@ -88,6 +81,13 @@ public HubMethodDescriptor(ObjectMethodExecutor methodExecutor, IServiceProvider
                     HasSyntheticArguments = true;
                     return false;
                 }
+                else if (serviceProviderIsService?.IsService(p.ParameterType) == true)
+                {
+                    Debug.Assert(index < 32);
```

Comment:
What if they have more than 32 parameters 😄

Reply:
Add a comment to the int above that this is used as a bitset.

Reply:
Is this actually enforced somewhere else?

Reply:
It should be shouldn't it? Or maybe just use a BitArray but that's pretty inefficient